addons/website/static/src/components/wysiwyg_adapter/wysiwyg_adapter.js

```diff
@@ -150,27 +150,19 @@ export class WysiwygAdapterComponent extends Wysiwyg {
             // Back navigation is handled with an additional state in the
             // history, used to capture the popstate event.
             history.pushState({ skipRouteChange: true }, '');
-            let hasFakeState = true;
             const leaveOnBackNavigation = () => {
-                hasFakeState = false;
                 this.leaveEditMode({
                     onStay: () => {
                         history.pushState({ skipRouteChange: true }, '');
-                        hasFakeState = true;
                     },
-                    onLeave: () => history.back(),
-                    reloadIframe: false
+                    onLeave: () => {},
+                    reloadIframe: true,
                 });
+                router.skipLoad = true;
             };
-            window.addEventListener('popstate', leaveOnBackNavigation);
```

Comment:
Why not keep popstate as it is to navigate back, what we should do is, we should add additional listener `BEFORE_ROUTE_CHANGE`, handler of this will do nothing but will change ```skipLoad``` to ```true```, rest of the things will remain as it is.